### ISSUE TITLE
MQTT documentation

### ISSUE THREAD

**Issue by Webbeh** [Feller-AG/wiser-tutorial#5]
Hallo !

I am developing a driver for Feller Wiser - Control4 operation, and I have a few questions. I don't yet have any devices with me to test so my questions might be dumb.

1. Is there no thermostat API ?
2. I saw that port 8883 must be opened. Are the devices communicating through MQTT ? If so, would it be ever possible to get the docs for the MQTT connection instead of the REST api ?
3. Is there anyone I can contact to get a testbench for me to be able to test my driver ?

Danke schön !

**Comment by swissmanu** [comment]
to know more about a potential mqtt protocol would be pretty awesome.

**Comment by woodworm** [comment]
1. There are no thermostats in the Wiser by Feller system yet.

2 .The MQTT connection is currently only designed for system-internal remote access.

3. For this question please contact the Feller Customer Service Center.

**Comment by swissmanu** [comment]
> 2 .The MQTT connection is currently only designed for system-internal remote access.

Are there any plans to open MQTT for third parties? The REST API is a good start for one-off commands. To synchronize device states across systems, polling via HTTP is cumbersome when there is already MQTT in place.

**Comment by woodworm** [comment]
Yes... it is planned to open MQTT for third parties.
In the meanwhile WebSocket could be an alternatieve solution for you. [see](https://github.com/Feller-AG/wiser-tutorial/blob/flumflum/demo/doc/websocket.md)

**Comment by Webbeh** [comment]
I guess that's an okay compromise for now, thanks!

**Comment by Stefacir** [comment]
> Yes... it is planned to open MQTT for third parties.
> In the meanwhile WebSocket could be an alternatieve solution for you. [see](https://github.com/Feller-AG/wiser-tutorial/blob/flumflum/demo/doc/websocket.md)

Hello together,
I am working on an integration into home assistant of Wiser right now.
Momentarily i work with Rest-API which works well for the basic functionality. 

But before i start to put a lot of work into the websocket functionality i‘d line to ask if you hace an estimate when MQTT will be released. 
It doesn‘t have to be very precisebut would be lovely to know if it‘s in the timerange of maybe the next year.

**Comment by woodworm** [comment]
It's part of our road map.

A first version next year is possible.
But some clarifications are still needed to find the right answer...
The answer to the ultimate question of what would we need to provide via Mqtt.
Your inputs are welcome... loads-state, button-events, sensor-events?

**Comment by Stefacir** [comment]
Thanks for your reply!
Sorry that it took so long for this answer.

I really appreciate that it could be possible to get an MQTT Version next year. That sounds great.
From my perspective it would be sufficient for a first version to have your mentioned. 
It should be possible to control all loads and get the states of the loads and as well the button/sensor events.

It's more or less what you are already providing by your WS API... :)

**Comment by sulazix** [comment]
Hi, I'm looking forward to seeing the public MQTT :-)

In the meantime, @Stefacir, did you manage to develop a working integration with HA? If so can you share it?

**Comment by Stefacir** [comment]
> Hi, I'm looking forward to seeing the public MQTT :-)
> 
> In the meantime, @Stefacir, did you manage to develop a working integration with HA? If so can you share it?

@sulazix
I didn't invested the time in developing in the HA integration as I wait until MQTT is released. Otherwise i'll do it unnecessary twice... 😉
What I did until now is integrating my devices via the REST API which works fine for me until MQTT is released.
Are you interested in this code as well?

**Comment by sulazix** [comment]
@Stefacir yeah with pleasure. If you don't want to create a repo only for this you Can send it by mail (my email is public on my profil) if you need any help next year for the HA integration I Can maybe help if you need/want... Not a lot of knowledge in HA yet but some expériences in python :-)

**Comment by moedele** [comment]
@Webbeh Do you finished and released some control4-driver somewhere?

@woodworm Whats the state of releasing mqtt to third parties?

**Comment by Webbeh** [comment]
Yes I did, under a commercial licence

**Comment by Webbeh** [comment]
> @Webbeh any link or connection on how to get to those drivers?

Through email: weby [at] we-bb [dot] com

**Comment by woodworm** [comment]
Unfortunately I can't provide a MQTT solution for third party yet because we're still working on other topics.
As soon as we have something I will report here.
